app/src/main/java/org/oppia/android/app/hintsandsolution/SolutionViewModel.kt

```diff
@@ -61,18 +65,29 @@ class SolutionViewModel private constructor(
   private val mathExpressionAccessibilityUtil: MathExpressionAccessibilityUtil,
   val explorationId: String,
   val isFlashback: Boolean,
-  val solutionBoxStrokeWidth: Int
+  val solutionBoxStrokeWidth: Int,
+  private val consoleLogger: ConsoleLogger
+
 ) {
-  /**
-   * A screenreader-friendly version of [solutionSummary] that should be used for readout, in place
-   * of the original summary.
+
+  /** Override default HTML parsing to include custom concept card tags when generating TalkBack-readable text for Solution and Flashback dialogs.
+   The previous implementation ignored custom tags, causing TalkBack to skip this fix ensures that TalkBack reads the concept card reference
+   exactly once and consistently.
    */
   val solutionSummaryContentDescription by lazy {
-    CustomHtmlContentHandler.fromHtml(
+    CustomHtmlContentHandler.getContentDescription(
       solutionSummary,
-      imageRetriever = null,
-      customTagHandlers = mapOf()
-    ).toString()
+      customTagHandlers = mapOf(
+        CUSTOM_CONCEPT_CARD_TAG to ConceptCardTagHandler(
+          listener = object : ConceptCardTagHandler.ConceptCardLinkClickListener {
+            override fun onConceptCardLinkClicked(view: View, skillId: String) {
+              // No action for TalkBack readout
+            }
```

Comment:
I think this could be:
```suggestion
          listener = conceptCardTagHandlerFactory.createConceptCardLinkClickListener()
```

Please try it out and let me know.

Reply:
Hi @adhiamboperes,

I tried applying the suggested change using
conceptCardTagHandlerFactory.createConceptCardLinkClickListener().

The build failed because conceptCardTagHandlerFactory was not previously injected into SolutionViewModel. While verifying this locally, I noticed that introducing the factory required updating multiple ViewModel constructors and their corresponding factory/create methods, which led to additional wiring changes across the codebase.

After reverting these changes, the app builds and runs correctly, and the TalkBack behavior remains unchanged. Since these broader wiring updates don’t seem strictly required for the accessibility fix itself, I reverted them to avoid introducing unnecessary changes.

Could you please confirm whether the factory-based listener change is required here, or if it’s better to keep the existing approach for now? I’m happy to update this further based on your guidance.

Thanks!